### PR TITLE
clean out ancient kbaseNarrativeWorkspace module (related to DATAUP-310)

### PR DESCRIPTION
**Pull request by briehl**
# Description of PR purpose/changes

So that state setup ticket is turning out to be a deeply nested onion. This module has the code that instantiates app cells (and viewer, editor, and other cells). It's also very ancient and had ~2000 lines of cruft that's either not used anywhere, or has no effect on cells made after 2017.  It wasn't hurting anything, so there wasn't much pressure to get rid of it.

Because this module controls setting up App cells, it should also be where Bulk Import cells get set up with the proper initial input data and app specs added to them, at least for consistency.

This first step is a pretty heavy dose of code deletion and cleanup. It's still probably incomplete, and feedback's welcome. I'll be adding tests next, though that might be challenging to get through in the scope of this sprint.

This can also be used as a starting point for setting up the next steps - initializing the Bulk Import cell with initial state and app spec information.

# Jira Ticket / Issue #
e.g. https://kbase-jira.atlassian.net/browse/DATAUP-X
- [x] Added the Jira Ticket to the title of the PR e.g. (DATAUP-69 Adds a PR template)

# Testing Instructions
* Details for how to test the PR: 
- [ ] Tests pass in travis and locally 
- N/A There should be no visible changes ~Changes available by spinning up a local narrative and navigating to _X_ to see _Y_~

# Dev Checklist:

- [ ] My code follows the guidelines at https://sites.google.com/truss.works/kbasetruss/development
- [ ] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: kbase-extension/static/kbase/js/widgets/narrative_core/kbaseNarrativeWorkspace.js [hunk 1]
```diff
@@ -1898,527 +577,6 @@ define([
...
-         * @param $logo - Target element
-         * @param type - Name of data type
-         * @param stacked - If true, show "stacked" version of the icon
-         *                indicating, e.g., that this is a container for
-         *                multiple items. Undefined is false.
-         * @param indent - Indent level (default is none)
-         */
-        setDataIcon: function ($logo, type, stacked, indent) {
-            if (indent === undefined || indent === null) {
-                indent = 0;
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/widgets/narrative_core/kbaseNarrativeWorkspace.js:2326]: Glad to see a copy of this code being deleted!

**Review by ialarmedalien — APPROVED** [review]
Delete away!

**Review by eamahanna — COMMENTED** [review]
How will these changes alter cells made before 2017? Is there a concern about backwards compatibility?

**Comment by briehl** [comment]
> How will these changes alter cells made before 2017? Is there a concern about backwards compatibility?

Those might actually go back to 2015 or earlier...

Those cells were the first and second iteration of what are our current app cells. They're no longer functional, and backwards compatibility has been handled there. Those cells that would be impacted get automatically converted on load to either an updated cell type or get turned into a markdown cell with an "obsolete! here's how to fix!" message.

**Comment by briehl** [comment]
Ok, I'm taking @ialarmedalien 's approval as an approval. :)

**Pull request merged**